platforms/documentation/docs/src/docs/userguide/reference/apis/kotlin_dsl.adoc

```diff
@@ -215,30 +225,70 @@ The Kotlin DSL currently provides various sets of type-safe model accessors, eac
 
 For the main project build scripts and precompiled project script plugins:
 
-* Dependency and artifact configurations (such as `implementation` and `runtimeOnly` contributed by the Java Plugin)
-* Project extensions and conventions (such as `sourceSets`), and extensions on them
-* Extensions on the `dependencies` and `repositories` containers, and extensions on them
-* Elements in the `tasks` and `configurations` containers
-* Elements in <<kotdsl:containers,project-extension containers>> (for example the source sets contributed by the Java Plugin that are added to the `sourceSets` container)
+[cols="~,~"]
+|===
+|Type-safe model accessors|Example
+
+|Dependency and artifact configurations
+|`implementation` and `runtimeOnly` (contributed by the Java Plugin)
+
+|Project extensions and conventions, and extensions on them
+|`sourceSets`
 
-For the main project settings script:
+|Extensions on the `dependencies` and `repositories` containers, and extensions on them
+|`testImplementation`, `mavenCentral`
 
-* Project extensions and conventions, contributed by `Settings` plugins, and extensions on them
+|Elements in the `tasks` and `configurations` containers
+|`compileJava`, `test`
+
+|Elements in <<kotdsl:containers,project-extension containers>>
+|Source sets contributed by the Java Plugin that are added to the `sourceSets` container: `sourceSets.main.java { setSrcDirs(listOf("src/main/java")) }`
+|===
+
+For the main project settings script and precompiled settings script plugins:
+
+[cols="~,~"]
+|===
+|Type-safe model accessors|Example
+
+|Project extensions and conventions, contributed by `Settings` plugins, and extensions on them
+|`pluginManagement`, `dependencyResolutionManagement`
+|===
 
 [IMPORTANT]
 ====
 Initialization scripts and script plugins do not have type-safe model accessors.
 These limitations will be removed in a future Gradle release.
 ====
 
-The set of type-safe model accessors available is calculated right before evaluating the script body, immediately after the `plugins {}` block.
-Any model elements contributed after that point do not work with type-safe model accessors.
-For example, this includes any configurations you might define in your own build script.
-However, this approach does mean that you can use type-safe accessors for any model elements that are contributed by plugins that are _applied by parent projects_.
+The set of type-safe model accessors available is determined right before evaluating the script body, immediately after the `plugins {}` block.
+Model elements contributed after that point, such as configurations defined in your build script, **will not work** with type-safe model accessors:
+
+[source,kotlin]
```

Comment:
:question: Please consider extracting this as a tested snippet in order for it not to rot in case we deprecate or break anything used in the snippet

Reply:
Done.